docs/concepts/validators.md

```diff
@@ -1,635 +1,575 @@
-* the second argument is the field value to validate; it can be named as you please
-* the third argument, if present, is an instance of `pydantic.ValidationInfo`
-* validators should either return the parsed value or raise a `ValueError` or `AssertionError` (``assert`` statements may be used).
-* A single validator can be applied to multiple fields by passing it multiple field names.
-* A single validator can also be called on *all* fields by passing the special value `'*'`.
+        1. Although `'invalid'` shouldn't validate against the `int` type, Pydantic accepts the input.
 
-!!! warning
-    If you make use of `assert` statements, keep in mind that running
-    Python with the [`-O` optimization flag](https://docs.python.org/3/using/cmdline.html#cmdoption-o)
-    disables `assert` statements, and **validators will stop working**.
+    === "Decorator"
 
-!!! note
-    `FieldValidationInfo` is **deprecated** in 2.4, use `ValidationInfo` instead.
+        ```python
+        from typing import Any
 
+        from pydantic import BaseModel, field_validator
 
-If you want to access values from another field inside a `@field_validator`, this may be possible using `ValidationInfo.data`, which is a dict of field name to field value.
-Validation is done in the order fields are defined, so you have to be careful when using `ValidationInfo.data` to not access a field that has not yet been validated/populated — in the code above, for example, you would not be able to access `info.data['id']` from within `name_must_contain_space`.
-However, in most cases where you want to perform validation using multiple field values, it is better to use `@model_validator` which is discussed in the section below.
 
-## Model validators
+        class Model(BaseModel):
+            number: int
 
-??? api "API Documentation"
-    [`pydantic.functional_validators.model_validator`][pydantic.functional_validators.model_validator]<br>
+            @field_validator('number', mode='plain')
+            @classmethod
+            def val_number(cls, value: Any) -> Any:
+                if isinstance(value, int):
+                    return value * 2
+                else:
+                    return value
 
-Validation can also be performed on the entire model's data using `@model_validator`.
 
-```python
-from typing import Any
+        print(Model(number=4))
+        #> number=8
+        print(Model(number='invalid'))  # (1)!
+        #> number='invalid'
+        ```
 
-from typing_extensions import Self
+        1. Although `'invalid'` shouldn't validate against the `int` type, Pydantic accepts the input.
 
-from pydantic import BaseModel, ValidationError, model_validator
+[](){#field-wrap-validator}
 
+- __*Wrap* validators__: they are the most flexible of all. You can run code before or after Pydantic and other validators
```

Comment:
Ok, so given that we are in the concepts section, I think this would be a great place to add a sentence or two explaining what the `handler` argument is (a callable that performs pydantic's internal validation, basically), and the fact that it must be specified as the second argument.